src/impl/tpls/KokkosBlas1_dot_tpl_spec_avail.hpp

```diff
@@ -93,10 +100,18 @@ Kokkos::View<const SCALAR*, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>, \
              Kokkos::MemoryTraits<Kokkos::Unmanaged> >, \
 1,1> { enum : bool { value = true }; };
 
+#if defined (KOKKOSKERNELS_INST_DOUBLE)
 KOKKOSBLAS1_DOT_TPL_SPEC_AVAIL_CUBLAS( double,                  Kokkos::LayoutLeft, Kokkos::CudaSpace)
+#endif
+#if defined (KOKKOSKERNELS_INST_FLOAT)
 KOKKOSBLAS1_DOT_TPL_SPEC_AVAIL_CUBLAS( float,                   Kokkos::LayoutLeft, Kokkos::CudaSpace)
+#endif
+#if defined (KOKKOSKERNELS_INST_KOKKOS_COMPLEX_DOUBLE_)
```

Comment:
How come the complex double and complex float macros end with underscores, but the double and float ones don't?

Reply:
I think we can use either without or with underscore, since in the ```KokkosKernels_config.h```, it uses
```#define KOKKOSKERNELS_INST_COMPLEX_DOUBLE
#if defined(KOKKOSKERNELS_INST_COMPLEX_DOUBLE)
#define KOKKOSKERNELS_INST_KOKKOS_COMPLEX_DOUBLE_
#endif```
 but I just followed the convention in hpp files in ```\src\impl\generated_specializations_hpp```

Reply:
OK, just checking :)